tracer/src/Datadog.Trace.MSBuild/DatadogLogger.cs

```diff
@@ -382,14 +380,25 @@ public override void Format(StringBuilder sb, LogFormatter formatter)
                     exception: null,
                     (JsonTextWriter writer, in Context? state) =>
                     {
-                        if (state.HasValue)
+                        if (state is { } context)
                         {
                             writer.WritePropertyName("dd.trace_id");
-                            writer.WriteValue($"{state.Value.TraceId}");
+
+                            if (context.TraceId.Upper > 0)
```

Comment:
It's probably worth checking with Tony, but should this be configured by the 128-bit logging configuration key?

Reply:
Yes, if these are logs that go into the Logs product, it should honor the config key. The reason we have the config key (and it defaults to disabled) is because the Logs backend can't handle 128-bit trace ids correctly yet.

Reply:
done